source/neuropod/bindings/java/BUILD

```diff
@@ -75,15 +97,46 @@ filegroup(
     ],
 )
 
-java_binary(
+# Unpack jar file in tmp directory, move files to expected subdirectories, pack to jar file again.
+# Note it runs inside sandbox where all created files and dirs are temporarily that deleted after build.
+genrule(
     name = "neuropod_java_jar",
+    srcs = [":neuropod_java_tmp.jar"],
+    outs = ["neuropod_java_jar.jar"],
+    cmd = select({
+        "@bazel_tools//src/conditions:darwin": """
+mkdir tmp
+cd tmp
+jar -xf ../$(location :neuropod_java_tmp.jar)
+jni_path="./com/uber/neuropod/native/darwin-x86_64"
+mkdir -p $$jni_path
+mv libneuropod_jni.dylib libneuropod.so neuropod_multiprocess_worker $$jni_path
+cd ..
+jar cf $@ -C tmp/ .  # Creates the new jar file.
+""",
+        "//conditions:default": """
+mkdir tmp
+cd tmp
+jar -xf ../$(location :neuropod_java_tmp.jar)
+jni_path="./com/uber/neuropod/native/linux-x86_64"
+mkdir -p $$jni_path
+mv libneuropod_jni.so libneuropod.so neuropod_multiprocess_worker $$jni_path
+cd ..
+jar cf $@ -C tmp/ .  # Creates the new jar file.
+""",
+    }),
+)
+
+java_binary(
+    name = "neuropod_java_tmp",
```

Comment:
We can entirely remove `neuropod_java_tmp` and just use `neuropod_java` in the genrule above.

(Since we're removing the `tar_backends` dependency here and the dependency on `backends` in `neuropod_java` below, the targets become identical)